pkg/api/v1beta3/dynakube/dynakube_types.go

```diff
@@ -0,0 +1,175 @@
+
+	// General configuration about OneAgent instances.
+	// You can't enable more than one module (classicFullStack, cloudNativeFullStack, hostMonitoring, or applicationMonitoring).
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OneAgent",xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	OneAgent OneAgentSpec `json:"oneAgent,omitempty"`
+
+	// Dynatrace apiUrl, including the /api path at the end. For SaaS, set YOUR_ENVIRONMENT_ID to your environment ID. For Managed, change the apiUrl address.
+	// For instructions on how to determine the environment ID and how to configure the apiUrl address, see Environment ID (https://www.dynatrace.com/support/help/get-started/monitoring-environment/environment-id).
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="API URL",order=1,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	APIURL string `json:"apiUrl"`
+
+	// Name of the secret holding the tokens used for connecting to Dynatrace.
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tenant specific secrets",order=2,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret"
+	Tokens string `json:"tokens,omitempty"`
+
+	// Adds custom RootCAs from a configmap. Put the certificate under certs within your configmap.
+	// Note: Applies to Dynatrace Operator, OneAgent and ActiveGate.
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Trusted CAs",order=6,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:io.kubernetes:ConfigMap"}
+	TrustedCAs string `json:"trustedCAs,omitempty"`
+
+	// Sets a network zone for the OneAgent and ActiveGate pods.
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Network Zone",order=7,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:com.tectonic.ui:text"}
+	NetworkZone string `json:"networkZone,omitempty"`
+
+	// Defines a custom pull secret in case you use a private registry when pulling images from the Dynatrace environment.
+	// To define a custom pull secret and learn about the expected behavior, see Configure customPullSecret
+	// (https://www.dynatrace.com/support/help/setup-and-configuration/setup-on-container-platforms/kubernetes/get-started-with-kubernetes-monitoring/dto-config-options-k8s#custompullsecret).
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Custom PullSecret",order=8,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:io.kubernetes:Secret"}
+	CustomPullSecret string `json:"customPullSecret,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	OpenTelemetryCollector OpenTelemetryCollectorSpec `json:"openTelemetryCollector,omitempty"`
```

Comment:
This element should be in a `templates` subsection, not directly on this level.

Reply:
fixed.